module/irrecoverable/unittest.go

```diff
@@ -4,44 +4,49 @@ import (
 	"context"
 	"testing"
 
-	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/mock"
+	"github.com/stretchr/testify/require"
 )
 
-// MockSignalerContext is a SignalerContext which will immediately fail a test if an error is thrown.
+// MockSignalerContext is a SignalerContext that can be used in tests to assert that an error is thrown.
+// It embeds a mock.Mock, so it can be used it to assert that Throw is called with a specific error.
+// Use NewMockSignalerContextExpectError to create a new MockSignalerContext that expects a specific error, otherwise NewMockSignalerContext.
 type MockSignalerContext struct {
 	context.Context
-	t           *testing.T
-	expectError error
+	*mock.Mock
 }
 
 var _ SignalerContext = &MockSignalerContext{}
 
 func (m MockSignalerContext) sealed() {}
 
 func (m MockSignalerContext) Throw(err error) {
-	if m.expectError != nil {
-		assert.EqualError(m.t, err, m.expectError.Error())
-		return
-	}
-	m.t.Fatalf("mock signaler context received error: %v", err)
+	m.Called(err)
 }
 
 func NewMockSignalerContext(t *testing.T, ctx context.Context) *MockSignalerContext {
-	return &MockSignalerContext{
+	m := &MockSignalerContext{
 		Context: ctx,
-		t:       t,
+		Mock:    &mock.Mock{},
 	}
+	m.Mock.Test(t)
+	t.Cleanup(func() { m.AssertExpectations(t) })
```

Comment:
nice! did you test to make sure the assertion was asserted when the error is never thrown

Reply:
Yes, I tested to ensure the assertion is asserted appropriately.